contrib/mesos/pkg/scheduler/service/service.go

```diff
@@ -390,11 +390,16 @@ func (s *SchedulerServer) prepareExecutorInfo(hks hyperkube.Interface) (*mesos.E
 		return nil, fmt.Errorf("either run this scheduler via km or else --executor-path is required")
 	} else {
 		if strings.Index(s.kmPath, "://") > 0 {
-			// URI could point directly to executable, e.g. hdfs:///km
-			// or else indirectly, e.g. http://acmestorage/tarball.tgz
-			// so we assume that for this case the command will always "km"
-			ci.Uris = append(ci.Uris, &mesos.CommandInfo_URI{Value: proto.String(s.kmPath), Executable: proto.Bool(true)})
-			ci.Value = proto.String("./km") // TODO(jdef) extract constant
+			if strings.HasPrefix(s.kmPath, "file://") {
+				// If `kmPath` started with "file://", `km` in agent local path was used.
+				ci.Value = proto.String(s.kmPath)
```

Comment:
actually I think I spoke too soon. The mesos code you're pointing at appears to be for handling URIs. Which is fine if you were copying kmPath into a URI. but here you're using it as the arg0 of a command line. so you'll want to strip the `file://` prefix upon assigning it to `ci.Value`.

again, have you tested this?